app/models/vm/operations/power.rb

```diff
@@ -1,4 +1,34 @@
 module Vm::Operations::Power
+  extend ActiveSupport::Concern
+
+  included do
+    supports :control do
+      msg = if retired?
+              _('The VM is retired')
+            elsif terminated?
+              _('The VM is terminated')
+            elsif !has_required_host?
+              _('The VM is not connected to a Host')
+            elsif !connection_state.nil? && !connected_to_ems?
+              _('The VM does not have a valid connection state')
+            elsif !has_active_ems?
+              _("The VM is not connected to an active #{ui_lookup(:table => "ext_management_systems")}")
+            end
+      unsupported_reason_add(:control, msg) if msg
+    end
+
+    supports :suspend do
+        msg = unsupported_reason(:control) unless supports_control?
+        msg ||= _('The VM is not powered on') unless vm_powered_on?
+        msg ||= _("Provider doesn't support vm_suspend") unless ext_management_system.respond_to?(:vm_suspend)
```

Comment:
ok, looks like we currently have no way of checking this dynamically.
so please remove this check, I'll take care of this shortcoming in a future PR